redirect/routes.php

```diff
@@ -262,20 +266,34 @@ class Route {
         ],
     ];
 
-    public function getTargetOfSub($subdomain) {
+    public function getTargetOfSub($subdomain, $subsub) {
+
         if ($subdomain === 'json') {
             header('Content-type: application/json');
             die(json_encode($this->routes));
         }
+
         if (isset($this->synonyms[ $subdomain ])) {
             $subdomain = $this->synonyms[ $subdomain ];
         }
 
+        switch ($subsub) {
+            case 'm' :  
+                // This is a moodle redirect like m.info1.tum.sexy
+                $moodle_id = $this->routes[ $subdomain ][ 'moodle_id' ];
+                if (!isset($moodle_id)) {
+                    return $this->routes[ $subdomain ][ 'target' ];  // Fallback to target if moodle id is unknown
+                }
+                return 'https://www.moodle.tum.de/course/view.php?id=' . $moodle_id;
+            default: 
+                break;
+        }
+
         if (!isset($this->routes[ $subdomain ])) {
             return 'http://tum.sexy/';
         }
 
-        return $this->routes[ $subdomain ]['target'];
+        return $this->routes[ $subdomain ][ 'target' ];
```

Comment:
Yuk, spaces. please remove.

Reply:
That's his the code was before :grinning: I just copied it and thought that's how PHP guys do it

Reply:
Not a valid excuse ;D